core/main/client/browser.js

```diff
@@ -109,7 +109,7 @@ beef.browser = {
      * @example: beef.browser.isIE10()
      */
     isIE10: function () {
-        return !!window.XMLHttpRequest && !window.chrome && !window.opera && !!document.documentMode && !window.XDomainRequest && !!window.performance && typeof navigator.msMaxTouchPoints !== "undefined";
+        return !!window.XMLHttpRequest && !window.chrome && !window.opera && !!document.documentMode && !!window.XDomainRequest && !!window.performance && typeof navigator.msMaxTouchPoints !== "undefined";
```

Comment:
@jcrew99 - !!/double-bang operator is effectively not, not. If something evaluates to nil, in this case window.XMLHttpRequest, !window.XMLHttpRequest will evaluate to false (because nil is falsy), and so !!window.XMLHttpRequest will equal true, because !false is the same as true.

Practically this means that we can use !!functionHere return true when a function doesn't exist in a browser. In this case, part of the IE10 fingerprinting is to check whether or not the window.XMLHttpRequest function exists. We need this check to evaluate to true because we want the function isIE10() to return true when that function is missing among a bunch of other checks. I'd suggest reading:

https://medium.com/@edplatomail/js-double-bang-or-the-not-operator-part-40e55d089bf0

Reply:
Sounds good, i wasn't sure if it was wanted to be true instead  of false. Thanks for the reading suggestion, good to go through!